pkg/encoder/encoder.go

```diff
@@ -0,0 +1,52 @@
+package encoder
```

Comment:
can we move to `plugin/internal/` or will other programs need to import this package?